9.0.2 `dependencyCheckAnalyze` fails with `NoSuchMethodError` Jackson `NativeImageUtil` method

Hi,

First of all many thanks for this great tool!

I recently upgraded the Dependency Check Gradle plugin to 9.0.2 and now the `dependencyCheckAnalyze` fails with `java.lang.NoSuchMethodError: 'boolean com.fasterxml.jackson.databind.util.NativeImageUtil.isInNativeImage()'` error.

The strange thing is that it does not happen always though - the failures are kind of intermittent.

This happens both locally and in CI build.

The NVD API data is downloaded successfully. The error occurs after it prints "Updating CISA Known Exploited Vulnerability list":

```
2023-12-04T11:01:47.1444429Z Completed processing batch 116/117 (99%) in 819ms
2023-12-04T11:01:47.9270340Z Completed processing batch 117/117 (100%) in 939ms
2023-12-04T11:01:48.1232257Z Updating CISA Known Exploited Vulnerability list: https://www.cisa.gov/sites/default/files/feeds/known_exploited_vulnerabilities.json
2023-12-04T11:01:48.5290707Z 
2023-12-04T11:01:48.5290734Z 
2023-12-04T11:01:48.5306217Z FAILURE: Build failed with an exception.
2023-12-04T11:01:48.5306782Z 
2023-12-04T11:01:48.5306969Z * What went wrong:
2023-12-04T11:01:48.5307964Z Execution failed for task ':sub-project:dependencyCheckAnalyze'.
2023-12-04T11:01:48.5309368Z > 'boolean com.fasterxml.jackson.databind.util.NativeImageUtil.isInNativeImage()'
2023-12-04T11:01:48.5310242Z 
2023-12-04T11:01:48.5310376Z * Try:
2023-12-04T11:01:48.5310924Z > Run with --debug option to get more log output.
2023-12-04T11:01:48.5311687Z > Run with --scan to get full insights.
2023-12-04T11:01:48.5312389Z > Get more help at https://help.gradle.org.
2023-12-04T11:01:48.5312879Z 
2023-12-04T11:01:48.5313034Z * Exception is:
2023-12-04T11:01:48.5314886Z org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':sub-project:dependencyCheckAnalyze'.
2023-12-04T11:01:48.5317114Z 	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.lambda$executeIfValid$1(ExecuteActionsTaskExecuter.java:148)
2023-12-04T11:01:48.5318900Z 	at org.gradle.internal.Try$Failure.ifSuccessfulOrElse(Try.java:282)
2023-12-04T11:01:48.5323126Z 	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeIfValid(ExecuteActionsTaskExecuter.java:146)
2023-12-04T11:01:48.5325421Z 	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:134)
2023-12-04T11:01:48.5327704Z 	at org.gradle.api.internal.tasks.execution.FinalizePropertiesTaskExecuter.execute(FinalizePropertiesTaskExecuter.java:46)
2023-12-04T11:01:48.5330081Z 	at org.gradle.api.internal.tasks.execution.ResolveTaskExecutionModeExecuter.execute(ResolveTaskExecutionModeExecuter.java:51)
2023-12-04T11:01:48.5332469Z 	at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:57)
2023-12-04T11:01:48.5334619Z 	at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:74)
2023-12-04T11:01:48.5336696Z 	at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:36)
2023-12-04T11:01:48.5338814Z 	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.executeTask(EventFiringTaskExecuter.java:77)
2023-12-04T11:01:48.5340820Z 	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:55)
2023-12-04T11:01:48.5342765Z 	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:52)
2023-12-04T11:01:48.5345374Z 	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:204)
2023-12-04T11:01:48.5348006Z 	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:199)
2023-12-04T11:01:48.5350232Z 	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:66)
2023-12-04T11:01:48.5352188Z 	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:59)
2023-12-04T11:01:48.5354162Z 	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:157)
2023-12-04T11:01:48.5356148Z 	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:59)
2023-12-04T11:01:48.5358228Z 	at org.gradle.internal.operations.DefaultBuildOperationRunner.call(DefaultBuildOperationRunner.java:53)
2023-12-04T11:01:48.5360345Z 	at org.gradle.internal.operations.DefaultBuildOperationExecutor.call(DefaultBuildOperationExecutor.java:78)
2023-12-04T11:01:48.5362373Z 	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter.execute(EventFiringTaskExecuter.java:52)
2023-12-04T11:01:48.5364233Z 	at org.gradle.execution.plan.LocalTaskNodeExecutor.execute(LocalTaskNodeExecutor.java:42)
2023-12-04T11:01:48.5366235Z 	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:331)
2023-12-04T11:01:48.5368587Z 	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:318)
2023-12-04T11:01:48.5371098Z 	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.lambda$execute$0(DefaultTaskExecutionGraph.java:314)
2023-12-04T11:01:48.5373322Z 	at org.gradle.internal.operations.CurrentBuildOperationRef.with(CurrentBuildOperationRef.java:80)
2023-12-04T11:01:48.5375499Z 	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:314)
2023-12-04T11:01:48.5378050Z 	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:303)
2023-12-04T11:01:48.5380200Z 	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.execute(DefaultPlanExecutor.java:463)
2023-12-04T11:01:48.5381968Z 	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.run(DefaultPlanExecutor.java:380)
2023-12-04T11:01:48.5383604Z 	at org.gradle.execution.plan.DefaultPlanExecutor.process(DefaultPlanExecutor.java:116)
2023-12-04T11:01:48.5385504Z 	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph.executeWithServices(DefaultTaskExecutionGraph.java:138)
2023-12-04T11:01:48.5387569Z 	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph.execute(DefaultTaskExecutionGraph.java:123)
2023-12-04T11:01:48.5389416Z 	at org.gradle.execution.SelectedTaskExecutionAction.execute(SelectedTaskExecutionAction.java:35)
2023-12-04T11:01:48.5391938Z 	at org.gradle.execution.DryRunBuildExecutionAction.execute(DryRunBuildExecutionAction.java:51)
2023-12-04T11:01:48.5394027Z 	at org.gradle.execution.BuildOperationFiringBuildWorkerExecutor$ExecuteTasks.call(BuildOperationFiringBuildWorkerExecutor.java:54)
2023-12-04T11:01:48.5396430Z 	at org.gradle.execution.BuildOperationFiringBuildWorkerExecutor$ExecuteTasks.call(BuildOperationFiringBuildWorkerExecutor.java:43)
2023-12-04T11:01:48.5398883Z 	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:204)
2023-12-04T11:01:48.5482151Z 	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:199)
2023-12-04T11:01:48.5484405Z 	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:66)
2023-12-04T11:01:48.5486819Z 	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:59)
2023-12-04T11:01:48.5488822Z 	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:157)
2023-12-04T11:01:48.5491004Z 	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:59)
2023-12-04T11:01:48.5492988Z 	at org.gradle.internal.operations.DefaultBuildOperationRunner.call(DefaultBuildOperationRunner.java:53)
2023-12-04T11:01:48.5494961Z 	at org.gradle.internal.operations.DefaultBuildOperationExecutor.call(DefaultBuildOperationExecutor.java:78)
2023-12-04T11:01:48.5497138Z 	at org.gradle.execution.BuildOperationFiringBuildWorkerExecutor.execute(BuildOperationFiringBuildWorkerExecutor.java:40)
2023-12-04T11:01:48.5499455Z 	at org.gradle.internal.build.DefaultBuildLifecycleController.lambda$executeTasks$10(DefaultBuildLifecycleController.java:313)
2023-12-04T11:01:48.5501584Z 	at org.gradle.internal.model.StateTransitionController.doTransition(StateTransitionController.java:266)
2023-12-04T11:01:48.5503564Z 	at org.gradle.internal.model.StateTransitionController.lambda$tryTransition$8(StateTransitionController.java:177)
2023-12-04T11:01:48.5505351Z 	at org.gradle.internal.work.DefaultSynchronizer.withLock(DefaultSynchronizer.java:44)
2023-12-04T11:01:48.5507100Z 	at org.gradle.internal.model.StateTransitionController.tryTransition(StateTransitionController.java:177)
2023-12-04T11:01:48.5509162Z 	at org.gradle.internal.build.DefaultBuildLifecycleController.executeTasks(DefaultBuildLifecycleController.java:304)
2023-12-04T11:01:48.5511443Z 	at org.gradle.internal.build.DefaultBuildWorkGraphController$DefaultBuildWorkGraph.runWork(DefaultBuildWorkGraphController.java:220)
2023-12-04T11:01:48.5513803Z 	at org.gradle.internal.work.DefaultWorkerLeaseService.withLocks(DefaultWorkerLeaseService.java:264)
2023-12-04T11:01:48.5515717Z 	at org.gradle.internal.work.DefaultWorkerLeaseService.runAsWorkerThread(DefaultWorkerLeaseService.java:128)
2023-12-04T11:01:48.5518122Z 	at org.gradle.composite.internal.DefaultBuildController.doRun(DefaultBuildController.java:181)
2023-12-04T11:01:48.5520050Z 	at org.gradle.composite.internal.DefaultBuildController.access$000(DefaultBuildController.java:50)
2023-12-04T11:01:48.5521986Z 	at org.gradle.composite.internal.DefaultBuildController$BuildOpRunnable.lambda$run$0(DefaultBuildController.java:198)
2023-12-04T11:01:48.5523893Z 	at org.gradle.internal.operations.CurrentBuildOperationRef.with(CurrentBuildOperationRef.java:80)
2023-12-04T11:01:48.5525752Z 	at org.gradle.composite.internal.DefaultBuildController$BuildOpRunnable.run(DefaultBuildController.java:198)
2023-12-04T11:01:48.5527631Z 	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
2023-12-04T11:01:48.5529418Z 	at org.gradle.internal.concurrent.AbstractManagedExecutor$1.run(AbstractManagedExecutor.java:47)
2023-12-04T11:01:48.5531703Z Caused by: java.lang.NoSuchMethodError: 'boolean com.fasterxml.jackson.databind.util.NativeImageUtil.isInNativeImage()'
2023-12-04T11:01:48.5533835Z 	at com.fasterxml.jackson.module.blackbird.BlackbirdModule.setupModule(BlackbirdModule.java:40)
2023-12-04T11:01:48.5535505Z 	at com.fasterxml.jackson.databind.ObjectMapper.registerModule(ObjectMapper.java:879)
2023-12-04T11:01:48.5574501Z 	at org.owasp.dependencycheck.data.update.cisa.KnownExploitedVulnerabilityParser.parse(KnownExploitedVulnerabilityParser.java:70)
2023-12-04T11:01:48.5576972Z 	at org.owasp.dependencycheck.data.update.KnownExploitedDataSource.update(KnownExploitedDataSource.java:82)
2023-12-04T11:01:48.5578546Z 	at org.owasp.dependencycheck.Engine.doUpdates(Engine.java:906)
2023-12-04T11:01:48.5579848Z 	at org.owasp.dependencycheck.Engine.initializeAndUpdateDatabase(Engine.java:711)
2023-12-04T11:01:48.5581255Z 	at org.owasp.dependencycheck.Engine.analyzeDependencies(Engine.java:637)
2023-12-04T11:01:48.5582821Z 	at org.owasp.dependencycheck.gradle.tasks.AbstractAnalyze.analyze(AbstractAnalyze.groovy:100)
2023-12-04T11:01:48.5584717Z 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
2023-12-04T11:01:48.5586504Z 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
2023-12-04T11:01:48.5588382Z 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
2023-12-04T11:01:48.5589983Z 	at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:125)
```

Versions:

- OWASP Dependency Check Gradle plugin: 9.0.2
- Gradle: 8.4 and 8.5
- JVM: 17.0.8 (Eclipse Adoptium 17.0.8+7)
- OS: Linux 6.6.3-1 (and Linux 6.2.0)

Could it be that another version of Jackson library is loaded somehow? 

Do you maybe have any idea what could cause this?
Many thanks in advance.

--- a/README.md
+++ b/README.md
@@ -10,6 +10,25 @@ known, published vulnerabilities.
 **Breaking Changes** are included in the 9.0.0 release. Please see the [9.0.0 Upgrade Notice](https://github.com/jeremylong/DependencyCheck#900-upgrade-notice)
 on the primary dependency-check site for more information.
 
+### Gradle Build Environment
+
+With 9.0.0 users may encounter issues with `NoSuchMethodError` exceptions due to
+dependency resolution. If you encounter this issue you will need to pin some of
+the transitive dependencies of dependency-check to specific versions. For example:
+
+/buildSrc/build.gradle
+```groovy
+dependencies {
+    constraints {
+        // org.owasp.dependencycheck needs at least this version of jackson. Other plugins pull in older versions..
+        add("implementation", "com.fasterxml.jackson:jackson-bom:2.16.0")
+        // org.owasp.dependencycheck needs these versions. Other plugins pull in older versions..
+        add("implementation", "org.apache.commons:commons-lang3:3.14.0")
+        add("implementation", "org.apache.commons:commons-text:1.11.0")
+    }
+}
+```
+
 ## Current Release
 
 The latest version is 

